lib/cc/engine/analyzers/violation.rb

```diff
@@ -8,8 +8,8 @@ module Analyzers
       class Violation
         attr_reader :issue
 
-        def initialize(base_points, issue, hashes)
-          @base_points = base_points
+        def initialize(language, issue, hashes)
+          @language = language
```

Comment:
The main issue for me is the inconsistency. We're passing in an object named `language_strategy` in `lib/cc/engine/analyzers/reporter.rb` but assigning it internally within Violation as `language`. We should probably stick with one name.

It seems like what we're passing in, based on the methods called on the object, is more like a strategy (or configuration) than a "language".